app/uk/gov/hmrc/thirdpartyapplication/repository/ApplicationRepository.scala

```diff
@@ -311,118 +349,130 @@ class ApplicationRepository @Inject() (mongo: ReactiveMongoComponent)(implicit v
       // Last Use Date
       case lastUsedBefore: LastUseBeforeDate => lastUsedBefore.toMongoMatch
       case lastUsedAfter: LastUseAfterDate   => lastUsedAfter.toMongoMatch
-      case _                                 => Json.obj() // Only here to complete the match
+      case _                                 => Document() // Only here to complete the match
     }
   }
 
-  private def convertToSortClause(sort: ApplicationSort): List[JsObject] = sort match {
-    case NameAscending         => List(sorting("name" -> 1))
-    case NameDescending        => List(sorting("name" -> -1))
-    case SubmittedAscending    => List(sorting("createdOn" -> 1))
-    case SubmittedDescending   => List(sorting("createdOn" -> -1))
-    case LastUseDateAscending  => List(sorting("lastAccess" -> 1))
-    case LastUseDateDescending => List(sorting("lastAccess" -> -1))
-    case NoSorting             => List()
-    case _                     => List(sorting("name" -> 1))
+  private def convertToSortClause(sort: ApplicationSort): List[Bson] = sort match {
+    case NameAscending         => List(Aggregates.sort(Sorts.ascending("name")))
+    case NameDescending        => List(Aggregates.sort(Sorts.descending("name")))
+    case SubmittedAscending    => List(Aggregates.sort(Sorts.ascending("createdOn")))
+    case SubmittedDescending   => List(Aggregates.sort(Sorts.descending("createdOn")))
+    case LastUseDateAscending  => List(Aggregates.sort(Sorts.ascending("lastAccess")))
+    case LastUseDateDescending => List(Aggregates.sort(Sorts.descending("lastAccess")))
+    case NoSorting             => List(Aggregates.sort(Document()))
```

Comment:
Is there any way to avoid this adding any sort for NoSorting?

Reply:
List() will work as it was before the migration.